flocker/control/_model.py

```diff
@@ -34,6 +47,8 @@ class DockerImage(object):
     :ivar unicode full_name: A readonly property which combines the repository
         and tag in a format that can be passed to `docker run`.
     """
+    repository = field(mandatory=True)
+    tag = field(mandatory=True, initial=u"latest")
```

Comment:
I've tried to minimize adding constraints so that I don't have to update a bazillion tests that pass in bytes or whatever.